### PR TITLE
[mds-agency] detect all-redundant telemetry

### PR DESCRIPTION
**Pull request by karcass**
return a 400 and add'l error info if no new telemetry was sent

### CODE REVIEW

File: packages/mds-agency/api.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ function api(app: express.Express): express.Express {
           // stash provider_id
           res.locals.provider_id = provider_id
 
-          log.info(providerName(provider_id), req.method, req.originalUrl)
+          // log.info(providerName(provider_id), req.method, req.originalUrl)
```
Comment (avatarneil) [inline packages/mds-agency/api.ts]: Should just delete

File: packages/mds-agency/api.ts [hunk 2]
```diff
@@ -727,16 +727,20 @@ function api(app: express.Express): express.Express {
     return null
   }
 
-  async function writeTelemetry(telemetry: Telemetry | Telemetry[]): Promise<void[]> {
+  async function writeTelemetry(telemetry: Telemetry | Telemetry[]) {
     if (!Array.isArray(telemetry)) {
-      const promises = [
+      const promises: (Promise<any>)[] = [
```
Comment (avatarneil) [inline packages/mds-agency/api.ts]: I'm gonna clone this down and see if I can find a better way to express the type than this hack we came up with earlier today... Seems super odd that we'd need to use the `Promise<any>[]` here. WE WILL REBUILD IT! MAKE IT STRONGER! FASTER!

Comment (mdurling) [reply packages/mds-agency/api.ts]: Try to use destructuring and let TS infer the types `const [x, y, z] = [...]`or declare the type as a tuple `const promises: [Promise<x>, Promise<y>, Promise<z>] = ...`

Comment (avatarneil) [reply packages/mds-agency/api.ts]:
```
async function writeTelemetry(telemetry: Telemetry | Telemetry[]) {
    if (!Array.isArray(telemetry)) {
      const promises: [Promise<number>, Promise<void>, Promise<void>] = [
        db.writeTelemetry([telemetry]),
        cache.writeTelemetry([telemetry]),
        stream.writeTelemetry([telemetry])
      ]
      return Promise.all(promises)
    }
    const promises: [Promise<number>, Promise<void>, Promise<void>] = [
      db.writeTelemetry(telemetry),
      cache.writeTelemetry(telemetry),
      stream.writeTelemetry(telemetry)
    ]
    return Promise.all(promises)
  }
```

seems to work!

File: packages/mds-agency/api.ts (hunk 1 above)
Comment (karcass) [reply packages/mds-agency/api.ts]: well, we had it before, we might want it again later
generally though I agree

File: packages/mds-agency/api.ts (hunk 2 above)
Comment (karcass) [reply packages/mds-agency/api.ts]: check it in!

**Pull request merged**